src/helper/cookie/index.ts

```diff
@@ -47,8 +71,27 @@ export const getSignedCookie: GetSignedCookie = async (c, secret, key?) => {
 }
 
 export const setCookie = (c: Context, name: string, value: string, opt?: CookieOptions): void => {
-  const cookie = serialize(name, value, { path: '/', ...opt })
-  c.header('set-cookie', cookie, { append: true })
+  // Cookie names prefixed with __Secure- can be used only if they are set with the secure attribute.
+  // Cookie names prefixed with __Host- can be used only if they are set with the secure attribute, must have a path of / (meaning any path at the host)
+  // and must not have a Domain attribute.
+  // Read more at https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Set-Cookie#cookie_prefixes'
+  const cookies = []
+  if (opt?.securePrefix) {
+    cookies.push(serialize('__Secure-' + name, value, { path: '/', secure: true, ...opt }))
+  }
+
+  if (opt?.hostPrefix) {
+    cookies.push(
+      serialize('__Host-' + name, value, { secure: true, path: '/', domain: undefined, ...opt })
```

Comment:
Do not change the secure, path, and domain attributes.
```suggestion
      serialize('__Host-' + name, value, { ...opt, secure: true, path: '/', domain: undefined })
```

Reply:
@Code-Hex what if a developer gives false as a value for `secure` with `prefix` value set to 'Secure'? This would be invalid. I think it would be better to override it in this case.

Reply:
@Datron Considering the specifications, when using a prefix, it's always necessary to set `secure` to true. Therefore, I believe the `secure: false` option should be ignored.

Reply:
This is implicit, so how about providing two option types to accept either one? What do you think, @yusukebe ?

like below:
```ts
type CookieOptions = {
  domain?: string
  expires?: Date
  httpOnly?: boolean
  maxAge?: number
  path?: string
  secure?: boolean
  signingSecret?: string
  sameSite?: 'Strict' | 'Lax' | 'None'
  partitioned?: boolean
} | {
  prefix: 'host',
  expires?: Date
  httpOnly?: boolean
  maxAge?: number
  signingSecret?: string
  sameSite?: 'Strict' | 'Lax' | 'None'
} | {
  prefix: 'secure',
  expires?: Date
  httpOnly?: boolean
  maxAge?: number
  path?: string
  signingSecret?: string
  sameSite?: 'Strict' | 'Lax' | 'None'
  domain?: string
}
```

Reply:
@Code-Hex perhaps we can keep the implicit behaviour and document this on hono.dev?

Reply:
I think we don't need to go so far and it is possible that the IDE will not do a good job of Types suggestion. Let's leave it as is.